source/scpd/quorum/QuorumIntersectionTests.d

```diff
@@ -0,0 +1,744 @@
+// Copyright 2016-2019 Stellar Development Foundation and contributors. Licensed
+// under the Apache License, Version 2.0. See the COPYING file at the root
+// of this distribution or at http://www.apache.org/licenses/LICENSE-2.0
+
+module scpd.quorum.QuorumIntersectionTests;
+
+import scpd.quorum.QuorumIntersectionChecker;
+import scpd.quorum.QuorumTracker;
+
+import scpd.Cpp;
+import scpd.types.Stellar_SCP;
+import scpd.types.Stellar_types : Hash, NodeID;
+import scpd.types.Utils;
+import scpd.types.XDRBase;
+
+import agora.common.Config;
+import agora.common.crypto.Key;
+
+const xvector!SCPQuorumSet nullvec;
+const shared_ptr!SCPQuorumSet nullquorum;
+
+// we replace the use of std::pair with a fixed-length array
+size_t first (size_t[2] pair) { return pair[0]; }
```

Comment:
This is one example where the code was changed from upstream. So we should definitely have a separate commit including all the C++ code, and then the conversion step commits after that.